onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -29,9 +30,19 @@
 
 
 def replace_special_characters_with_underscores(data):
+    # Remove metadata fields from the column headers
+    remove_metadata_fields(data)
```

Comment:
Let's leave this out of `replace_special_characters_with_underscores`. _Keeping it separate is perfect, initially thought this function was supposed to clean the data_